webUI/react/src/components/reader/pager/VerticalPager.tsx

```diff
@@ -29,36 +29,87 @@ export default function VerticalReader(props: IReaderProps) {
     const classes = useStyles();
     const history = useHistory();
 
-    const [initialScroll, setInitialScroll] = useState(-1);
-    const initialPageRef = useRef<HTMLDivElement>(null);
+    const pageRef = useRef<HTMLDivElement>(null);
+    const pagesRef = useRef<HTMLDivElement[]>([]);
+    let readerCurPage = curPage;
+
+    function nextPage() {
+        if (curPage < pages.length - 1) {
+            setCurPage((x) => {
+                readerCurPage = x + 1;
+                return readerCurPage;
+            });
+            pagesRef.current[readerCurPage].scrollIntoView();
+        } else if (settings.loadNextonEnding) {
+            nextChapter();
+        }
+    }
+
+    function prevPage() {
+        if (readerCurPage > 0) {
+            setCurPage((page) => {
+                const rect = pagesRef.current[page].getBoundingClientRect();
+                readerCurPage = (rect.y < 0 && rect.y + rect.height > 0) ? page : page - 1;
+                return readerCurPage;
+            });
+        }
+        pagesRef.current[readerCurPage].scrollIntoView();
+    }
+
+    function keyboardControl(e:KeyboardEvent) {
+        switch (e.key) {
```

Comment:
use https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/code
because `case ' '` is a bit confusing to read.